### ISSUE TITLE
update lsp-mode over TRAMP docs

### ISSUE THREAD

**Issue by saltsucker** [emacs-lsp/lsp-mode#3841]
Is your feature related or already mentioned on [the wishlist](https://github.com/emacs-lsp/lsp-mode/discussions/2892)? No.

Can you please update the LSP-mode over tramp tutorial? It is sorely unfulfilling for anyone that wants to try to make this work. There are so many broken semi-solutions all over the internet of people basically shooting in the dark and crossing their fingers. I would make a PR, but... I can't make it work no matter what I try, so I can't contribute.

Some examples of more helpful info:
What OS's are known to work?
What remote OS's are known to work?
What is the *exact* environment required locally and remotely? Beside just "install LSP on local and remote"?
Does it work from Windows Subsystem for Linux?
What if you're using pyenv? (or some other python virtual environment tool)

Basically having an actually fulfilling fleshed out example would just be way more helpful.

https://emacs-lsp.github.io/lsp-mode/page/remote/

**Comment by yyoncho** [comment]
tramp support will be rewritten and will work OOTB with little/no configuration. Back then when I implemented tramp support it had tons of bizzare issues(e. g. for some os configurations it converts \n to \r).

**Comment by yyoncho** [comment]
@saltsucker it will be soon. But IMHO the overall experience won't be good anyway if you want to use it for serious work it may not perform well(due to blocking IO). FWIW if you want to connect to docker container there are alternatives.

**Comment by saltsucker** [comment]
> @saltsucker it will be soon. But IMHO the overall experience won't be good anyway if you want to use it for serious work it may not perform well(due to blocking IO). FWIW if you want to connect to docker container there are alternatives.

Ya... I had a feeling it was going to be a laggy experience (if that's what you mean). But VSCode can do it well (iirc, haven't used it in a while). Does that have non-blocking IO? 

I am not familiar with the docker way, so I am not sure what you mean. I want TRAMP so I can remote into my work servers. Navigate them, work on projects, etc. How does docker fit into that workflow? Spinning up a docker container on work server and trying to TRAMP into that from my PC doesn't seem better.

What other alternatives can I try? I think I tried sshfs and I don't think that worked well either.

**Comment by cpoile** [comment]
@yyoncho Was there any success in getting this to work? (Appreciate your help!)

**Comment by yyoncho** [comment]
@cpoile I am working on this right now.
